radix-engine/src/transaction/toolkit_transaction_receipt/receipt/base.rs

```diff
@@ -0,0 +1,142 @@
+        state_updates_summary: StateUpdatesSummary<T>,
+        /// The instruction-by-instruction worktop updates that took place in the transaction.
+        worktop_changes: IndexMap<T::Usize, Vec<T::WorktopChange>>,
+        /// The summary of the fees paid in the transaction.
+        fee_summary: FeeSummary<T>,
+        /// Information about the fees locked by the transaction.
+        locked_fees: LockedFees<T>,
+    },
+    /// The transaction would've been committed failure if it were submitted
+    /// to the ledger.
+    CommitFailure {
+        /// A [`String`] of the Rust debug formatted runtime error that caused the transaction to
+        /// fail. This is not a JSON string or object since we do not currently offer any guarantees
+        /// on the structure of the returned string.
+        reason: String,
+    },
+    /// The transaction would've been rejected if it were submitted to the
+    /// ledger.
+    Reject {
+        /// A [`String`] of the Rust debug formatted runtime error that caused the transaction to
+        /// be rejected. This is not a JSON string or object since we do not currently offer any
+        /// guarantees on the structure of the returned string.
+        reason: String,
+    },
+    /// The transaction would've been aborted if it were submitted to the
+    /// ledger.
+    Abort {
+        /// A [`String`] of the Rust debug formatted runtime error that caused the transaction to
+        /// be aborted. This is not a JSON string or object since we do not currently offer any
+        /// guarantees on the structure of the returned string.
+        reason: String,
+    },
+}
+
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "T::Decimal: Serialize + DeserializeOwned")]
+pub struct FeeSummary<T: TypeSelector> {
+    pub execution_fees_in_xrd: T::Decimal,
+    pub finalization_fees_in_xrd: T::Decimal,
+    pub storage_fees_in_xrd: T::Decimal,
+    pub royalty_fees_in_xrd: T::Decimal,
+}
+
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "T::Decimal: Serialize + DeserializeOwned")]
+pub struct LockedFees<T: TypeSelector> {
+    pub contingent: T::Decimal,
+    pub non_contingent: T::Decimal,
+}
+
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "T::MetadataValue: Serialize + DeserializeOwned")]
+pub enum MetadataUpdate<T: TypeSelector> {
+    Set(T::MetadataValue),
+    Delete,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "
+        T::Bytes: Serialize + DeserializeOwned,
+        T::NodeId: Serialize + DeserializeOwned,
+        T::Decimal: Serialize + DeserializeOwned,
+        T::MetadataValue: Serialize + DeserializeOwned,
+        T::NonFungibleGlobalId: Serialize + DeserializeOwned,
+    ")]
+pub struct StateUpdatesSummary<T: TypeSelector> {
+    /// The set of new entities created in this transaction.
+    pub new_entities: IndexSet<T::NodeId>,
```

Comment:
@dhedey This is different from my original proposal. The following is the reason for it:

> With this new receipt, the toolkit is actually pivoting to a completely new approach of getting the data that it needs. Previously, the toolkit used the execution trace to compute the worktop changes, the state updates summary to compute the set of newly created entities, and the state updates to compute the updated non-fungible data and metadata. This meant that the toolkit required that the execution trace, state update summary, and state updates be provided in the transaction receipt that it was given and that the toolkit imposed what the source of the data was. This mean that any structural changes to the data sources (e.g., a new field added to the execution trace, or worktop changes computation being made possible through events instead of the execution trace) would be a breaking change to the toolkit. However, the toolkit really does not need or benefit in any way by imposing how the data it needs is derived, all that it actually needs is the data. Therefore, with the updated approach, the toolkit receipt doesn't (and should not) expose the state updates, execution trace, application events, or anything else of that sort. The computation or required interpretation is done in the core-api or the engine which gives us more freedom on how this data is computed.